docs/content/advanced/backend/upgrade-guides/1-7-1.md

```diff
@@ -0,0 +1,113 @@
+---
+description: 'Actions Required for v.1.7.1'
+---
+
+# v1.7.1
+
+Version `1.7.1` of Medusa introduces the `JobSchedulerService` which changes how scheduled/cron jobs are created.
+
+## Overview
+
+Version `1.7.1` of Medusa introduces a new service `JobSchedulerService` that handles all logic and functionality related to created scheduled (previously named cron jobs).
+
+With this introduction, the previous use of `EventBus` to create a cron job has been deprecated of using the `JobSchedulerService`.
+
+In addition, this version features some fixes to gift cards that requires running migrations, and changes to how payment providers are implemented.
+
+## Actions Required
+
+### Run Migrations
+
+In the directory of your Medusa server, run the following command after updating the server:
+
+```bash
+medusa migrations run
+```
+
+### Run Migration Script
+
+Following the fix gift cards calculation, you also need to run a migration script after updating the server.
+
+Start by adding the following environment variables in `.env`:
+
+```bash
+TYPEORM_CONNECTION=postgres
+TYPEORM_URL=<DATABASE_URL>
+TYPEORM_USERNAME=<DATABASE_USERNAME>
+TYPEORM_PASSWORD=<DATABASE_PASSWORD>
+TYPEORM_DATABASE=<DATABASE_DATABASE>
+TYPEORM_ENTITIES=./node_modules/@medusajs/medusa/dist/models/*.js
+TYPEORM_MIGRATIONS=./node_modules/@medusajs/medusa/dist/migrations/*.js
+```
+
+Make sure to replace `<DATABASE_URL>`, `<DATABASE_USERNAME>`, `<DATABASE_PASSWORD>`, and `<DATABASE_DATABASE>` with your database connection details.
+
+Then, run the following command in the root directory of your Medusa server:
+
+```bash
+node ./node_modules/@medusajs/medusa/dist/scripts/gift-card-tax-rate-migration.js
+```
+
+### Change to JobSchedulerService
+
+In your loader file that creates a cron job, replace the use of `eventBus` to `jobSchedulerService`:
+
+```ts
+const myJob = async (container, options) => {
+  const jobSchedulerService = container.resolve("jobSchedulerService");
+  jobSchedulerService.create("my-job", {}, "0 0 * * *", async () => {
+    //...
+  })
```

Comment:
suggestion(nit): I realize the intention here is to show the difference in using the services, but I would still add the default export just in case something does a c/p of the code snippet

Reply:
Good point! I added it.